src/ILCompiler.CppCodeGen/src/CppCodeGen/CppWriter.cs

```diff
@@ -1268,18 +1235,8 @@ private String GetCodeForReloc(Relocation reloc, DependencyNode node, NodeFactor
 
                 bool isEagerCctorTable = symbolNode.Target is ArrayOfEmbeddedPointersNode<IMethodNode>;
 
-                if ((symbolNode.Target is GenericVirtualMethodTableNode ||
-                    symbolNode.Target is InterfaceGenericVirtualMethodTableNode ||
-                    symbolNode.Target is ExactMethodInstantiationsNode ||
-                    symbolNode.Target is GenericMethodsHashtableNode ||
-                    symbolNode.Target is NativeLayoutInfoNode ||
-                    symbolNode.Target is MetadataNode ||
-                    symbolNode.Target is BlockReflectionTypeMapNode ||
-                    symbolNode.Target is ExternalReferencesTableNode ||
-                    symbolNode.Target is GenericTypesTemplateMap ||
-                    symbolNode.Target is GenericMethodsTemplateMap ||
-                    symbolNode.Target is GenericTypesHashtableNode ||
-                    symbolNode.Target is TypeMetadataMapNode ||
+                if ((!(symbolNode.Target is EmbeddedDataContainerNode) &&
+                    !(symbolNode.Target is StackTraceMethodMappingNode) ||
                     isEagerCctorTable
```

Comment:
`isEagerCctorTable` condition does not seem to be necessary now that the logic is flipped.

Reply:
`isEagerCctorTable` is needed because `!(symbolNode.Target is EmbeddedDataContainerNode)` filters all `ArrayOfEmbeddedPointersNode` nodes.